internal/internal_task_pollers.go

```diff
@@ -265,6 +266,7 @@ func (bp *basePoller) getCapabilities() *workflowservice.GetSystemInfoResponse_C
 // newWorkflowTaskPoller creates a new workflow task poller which must have a one to one relationship to workflow worker
 func newWorkflowTaskPoller(
 	taskHandler WorkflowTaskHandler,
+	contextManager WorkflowContextManager,
```

Comment:
Doesn't the `taskHandler` imply the `contextManager`? And if so, do you really need a separate interface and field here?

Reply:
You're correct. What we ran into here is the (arguable) fact that ContextManager should probably be a concept independent from the TaskHandler - thus the two interfaces. But the reality is that context management is currently highly coupled to this one specific workflowTaskHandlerImpl - the context objects call back into that task handler to access the cache (direct field access no less) among other things.

Actual separation of the implementations was just going to be way too much churn for this PR.